plugins/pulp_rpm/plugins/distributors/yum/publish.py

```diff
@@ -841,7 +841,14 @@ def _write_pulp_distribution_file(self, distro_files, old_xml_file_path):
             element.text = distro_file
 
         tree = cElementTree.ElementTree(new_xml_root)
-        tree.write(os.path.join(self.get_working_dir(), constants.DISTRIBUTION_XML))
+        distribution_xml_path = os.path.join(self.get_working_dir(), constants.DISTRIBUTION_XML)
+        try:
+            os.remove(distribution_xml_path)
+        except OSError as e:
+            if e.errno != os.errno.ENOENT:
+                raise
+
+        tree.write(distribution_xml_path)
```

Comment:
You should leave off the argument here. Using ``raise`` alone re-raises the current exception being handled, preserving its scope. Passing an argument here replaces that scope, which ends up replacing the original traceback with one that starts here.

You can see the difference in this simple example:

```
>>> def foo():
...  raise ValueError
... 
>>> foo()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "<stdin>", line 2, in foo
ValueError
>>> try:
...  foo()
... except Exception as e:
...  raise e
... 
Traceback (most recent call last):
  File "<stdin>", line 4, in <module>
ValueError
>>> try:
...  foo()
... except Exception as e:
...  raise
... 
Traceback (most recent call last):
  File "<stdin>", line 2, in <module>
  File "<stdin>", line 2, in foo
ValueError
```